documentation/decisions/adr/000N-frontend-tooling.md

```diff
@@ -0,0 +1,82 @@
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- **Ease of use and configurability:** Code quality tools should be configurable to meet the specific needs of our project and enforce standards uniformly. Running the tool should be simple, and feedback should be easily implementable and understandable.
+- **Speed:** Code quality tools should be able to run in an timely manner, ideally in a pre-commit hook.
+- **Documentation and resources:** Code quality tools should have robust and helpful documentation, specifically around usage and configuration.
+- **Lean:** Selection of code quality tools should cover all use-cases with little overlap for ease of use and to minimize mental surface area required to understand the system. As in, each tool should have a specific, meaningful purpose.
+- **Enforces security:** Tooling should assist us in identifying security concerns and vulnerabilities.
+- **Well-maintained:** We have a preference towards libraries that are widely adopted and have active maintainers.
+
+## Options Considered
+
+1. Use a curated collection of JavaScript libraries from Next.js template repository (described below)
+2. Use a different or modified set of JavaScript libraries
+
+### Dependency Management
+
+**[npm](https://www.npmjs.com/):** Package manager for Node.js.
+- npm is a robust package manager that is pre-bundled with Node, simplifying installation steps
+- Maintained by Github/Microsoft
+
+### Code Linting
+
+**[ESLint](https://eslint.org/):** Statically analyzes your code to quickly find problems.
+- Supported by a huge variety of editors and easy to integrate into CI/CD pipeline
+- Very long-standing project, hugely popular, and well-supported tool
+
+### Auto-formatting
+
+**[Prettier](https://prettier.io/):** Enforces code style and consistency while reducing the need for manual formatting or nitpicking.
+- Top pick in many JavaScript repositories as a well-maintained formatting tool
+
+### Type Checking
+
+**[TypeScript](https://www.typescriptlang.org/):** Strongly typed programming language that builds on JavaScript
+- Converts to JavaScript, therefore works everywhere JavaScript is compatible
+- Static-type checking build into language
+- Identifies problems early at compile time
+- Open source and maintained by Microsoft
+
+### Dependency Checks
+
+**[Renovate](https://docs.renovatebot.com/):** Automated dependency updates.
+- Free and open source
+- Can batch, schedule, auto-assign and auto merge updates
+- Multi-language and supports Python as well
+
+*Note: Do we need the Safety package for our Python code as well or is this sufficient?*
+
+### License Checking[^*]
+**[License Checker](https://www.npmjs.com/package/license-checker):** Very aptly named tool to check licenses for dependencies.
```

Comment:
Having a bit of trouble finding a license checking tool that will work well with automation. I'll put some more time into it, but let me know if you have any ideas!